### ISSUE TITLE
Create new squashed migration for wagtailcore

### ISSUE THREAD

**Issue by gasman** [wagtail/wagtail#4965]
#4916 (creating a new squashed migration for wagtailcore) was reverted for the 2.4 release (in #4963) because it caused `./manage.py makemigrations` to fail on a newly-upgraded Wagtail installation, as reported in #4961.

It's not clear why this was failing - it's most likely that we need some special treatment for the fact that we're replacing an existing squashed migration, but in my tests it was failing in the same way regardless of whether the `django_migrations` table contained an entry for the old squashed migration, and I wasn't able to reproduce this independently on a new Django project. Need to investigate this further, and fix if possible.

**Comment by gasman** [comment]
https://docs.djangoproject.com/en/2.2/topics/migrations/#squashing-migrations makes it clear that we've been doing squashed migrations wrong: you're not meant to keep the original migrations around indefinitely, and you're definitely not meant to do another round of squashing until you've removed them. However, you're also not meant to remove them until you're sure that all installations of your site have migrated past the squash point - which isn't really feasible for a reusable project like Wagtail.

(You might argue that we do expect users to upgrade one version at a time - makemigrations is liable to do the wrong thing otherwise - and so it would be fair game for Wagtail version N to remove the unsquashed migrations from Wagtail N-1, on the basis that they should have already migrated past that point. However, this would also require devs to _deploy_ after every point upgrade to ensure that their production instances are migrated past the migrations that have been removed, and I don't think that's reasonable: a dev should be able to run through all the stepwise upgrades locally and deploy them all in one go.)

Basically: squashing our migrations is going to cause way more headaches than a few more seconds of build time ever will. Let's not do it again. Ever. :-)
